File: Chapter08/HelpDesk/HelpDesk/HelpDesk.Server/DataSources/ApplicationData/_ApplicationDataService.lsml.vb
Namespace LightSwitchApplication
    Public Class ApplicationDataService

        Private Sub Issues_Validate(
  entity As Issue, results As EntitySetValidationResultsBuilder)

            'Listing 7-5. Validating the counts of child items 
            If entity.IssueDocuments.Count() > 10 Then
                results.AddEntityError(
      "Issues can only contain a maximum of 10 documents")
            End If


            'Listing 7-10. Validating Deletions on the Server 

            ' Check for validation errors for deletions
            If entity.Details.EntityState = EntityState.Deleted Then
                If entity.IssueResponses.Where(
      Function(s) s.AwaitingClient).Any() Then
                    results.AddEntityError(
           "Cannot delete issues with responses awaiting client.")
                End If
            End If
        End Sub

        Private Sub Issues_Deleting(entity As Issue)

            ' Check for validation errors for deletions
            If entity.Details.ValidationResults.Errors.Any Then
                Throw New ValidationException(Nothing,
                  Nothing, entity.Details.ValidationResults)
            End If

            ' Cascade delete children because delete rule is Restricted
            For Each resp In entity.IssueResponses
                resp.Delete()
            Next


        End Sub


    End Class
End Namespace

File: Chapter07/HelpDeskCS/HelpDeskCS/HelpDeskCS.Server/DataSources/ApplicationData/_ApplicationDataService.lsml.cs
using Microsoft.LightSwitch.Security.Server;
using Microsoft.LightSwitch;
using System.Text;
using System.Linq;
using System.Collections.Generic;
using System;

namespace LightSwitchApplication
{
    public partial class ApplicationDataService
    {

        //Listing 7-5. Validating the counts of child items

        partial void Issues_Validate(Issue entity, EntitySetValidationResultsBuilder results)
        {
            if (entity.IssueDocuments.Count() > 10)
            {
                results.AddEntityError(
                  "Issues can only contain a maximum of 10 documents");
            }


            //Listing 7-10. Validating deletions on the server

            if (entity.Details.EntityState == EntityState.Deleted)
            {
                if (entity.IssueResponses.Where(s => s.AwaitingClient).Any())
                {
                    results.AddEntityError(
                      "Cannot delete issues with responses awaiting client.");
                }
            }


        }

        //Listing 7-10. Validating deletions on the server
        partial void Issues_Deleting(Issue entity)
        {
            // Check for validation errors for deletions
            if (entity.Details.ValidationResults.Errors.Any())
            {
                throw new ValidationException(
                  null, null, entity.Details.ValidationResults);
            }

            // Cascade delete children because delete rule is Restricted
            foreach (var childResp in entity.IssueResponses)
            {
                childResp.Delete();
            }
        }





    }
}